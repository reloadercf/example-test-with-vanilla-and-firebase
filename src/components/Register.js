import { onNavigate } from '../main.js';
import { addUser } from '../lib/auth.js';

export const Register = () => {
  const div = document.createElement('div');
  const title = document.createElement('h2');
  const button = document.createElement('button');
  const buttonBack = document.createElement('button');
  const inputEmail = document.createElement('input');
  const inputPass = document.createElement('input');
  const mensaje = document.createElement('p');

  button.textContent = 'Crear cuenta';
  button.classList.add('registro');

  buttonBack.textContent = 'Regresa';
  buttonBack.classList.add('Regresa');
  title.textContent = 'Registro de nueva cuenta';

  inputEmail.classList.add('correo');
  inputPass.classList.add('pass');
  mensaje.classList.add('mensaje');

  button.addEventListener('click', () => {
    addUser(inputEmail.value, inputPass.value).then((userCredential) => {
      // Signed in
      const textNode = document.createTextNode('se registro');
      mensaje.append(textNode);
      const user = userCredential.user;
      console.log('welcome ', user);
      // ...
    })
      .catch((error) => {
        mensaje.textContent = 'no se registro';
        const errorCode = error.code;
        const errorMessage = error.message;
        console.log(error);
      // ..
      });
  });
  buttonBack.addEventListener('click', () => {
    onNavigate('/');
  });

  div.append(title, inputEmail, inputPass, mensaje, button, buttonBack);

  return div;
};
