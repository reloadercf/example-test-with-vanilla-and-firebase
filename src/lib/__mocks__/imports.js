const user = {
  userCredential: {
    uid: 'kj2h3kjrhn2k3jh423',
    email: 'carlos@carlos.com',
  },
};
export const createUserWithEmailAndPassword = jest.fn(() => Promise.resolve({ user }));
export const getAuth = () => user;
export const initializeApp = () => Promise.resolve({});
