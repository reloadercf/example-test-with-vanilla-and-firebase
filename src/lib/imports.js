export { getAuth, createUserWithEmailAndPassword } from 'https://www.gstatic.com/firebasejs/9.10.0/firebase-auth.js';
export { initializeApp } from 'https://www.gstatic.com/firebasejs/9.10.0/firebase-app.js';
