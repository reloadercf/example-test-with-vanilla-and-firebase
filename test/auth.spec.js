/**
 * @jest-environment jsdom
 */

import { Register } from '../src/components/Register.js';
import { createUserWithEmailAndPassword } from '../src/lib/__mocks__/imports.js';

jest.mock('../src/main.js');
jest.mock('../src/lib/imports.js');

describe('Los test del Registro', () => {
  test('El componente es una Funcion', () => {
    expect(typeof Register).toBe('function');
  });
  test('Existe el boton de crear cuenta', () => {
    const elemento = Register();
    const boton = elemento.querySelector('.registro');
    expect(boton).not.toBeNull();
  });
  test('Snapshot del REGISTRO', () => {
    expect(Register()).toMatchSnapshot();
  });

  test('Invoca a la promesa con los argumentos escritos ', () => {
    const elemento = Register();
    const email = elemento.querySelector('.correo');
    const pass = elemento.querySelector('.pass');
    const boton = elemento.querySelector('.registro');

    email.value = 'alguien@correo.com';
    pass.value = '123456';

    boton.addEventListener('click', () => {
      createUserWithEmailAndPassword(email.value, pass.value);
    });
    boton.click();
    boton.dispatchEvent(new Event('click'));
    expect(createUserWithEmailAndPassword).toHaveBeenCalled();
    expect(createUserWithEmailAndPassword).toHaveBeenLastCalledWith('alguien@correo.com', '123456');
  });
});
