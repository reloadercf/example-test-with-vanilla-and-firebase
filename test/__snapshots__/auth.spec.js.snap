// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Los test del Registro Snapshot del REGISTRO 1`] = `
<div>
  <h2>
    Registro de nueva cuenta
  </h2>
  <input
    class="correo"
  />
  <input
    class="pass"
  />
  <p
    class="mensaje"
  />
  <button
    class="registro"
  >
    Crear cuenta
  </button>
  <button
    class="Regresa"
  >
    Regresa
  </button>
</div>
`;
